src/gui/draw.h

```diff
@@ -259,34 +260,61 @@ static inline void dt_draw_curve_set_point(dt_draw_curve_t *c, const int num, co
   c->c.m_anchors[num].y = y;
 }
 
-static inline void dt_draw_curve_calc_values(dt_draw_curve_t *c, const float min, const float max,
-                                             const int res, float *x, float *y)
+static inline void dt_draw_curve_smaple_values(dt_draw_curve_t *c, const float min, const float max, const int res,
+                                               float *x, float *y)
 {
-  c->csample.m_samplingRes = res;
-  c->csample.m_outputRes = 0x10000;
-  CurveDataSample(&c->c, &c->csample);
   if(x)
   {
 #ifdef _OPENMP
-#pragma omp parallel for SIMD() default(none) \
-    dt_omp_firstprivate(res) \
-    shared(x) \
-    schedule(static)
+#pragma omp parallel for SIMD() default(none) dt_omp_firstprivate(res) shared(x) schedule(static)
 #endif
     for(int k = 0; k < res; k++) x[k] = k * (1.0f / res);
   }
   if(y)
   {
 #ifdef _OPENMP
-#pragma omp parallel for SIMD() default(none) \
-    dt_omp_firstprivate(min, max, res) \
-    shared(y, c) \
-    schedule(static)
+#pragma omp parallel for SIMD() default(none) dt_omp_firstprivate(min, max, res) shared(y, c) schedule(static)
 #endif
     for(int k = 0; k < res; k++) y[k] = min + (max - min) * c->csample.m_Samples[k] * (1.0f / 0x10000);
   }
 }
 
+static inline void dt_draw_curve_calc_values(dt_draw_curve_t *c, const float min, const float max, const int res,
+                                             float *x, float *y)
+{
+  c->csample.m_samplingRes = res;
+  c->csample.m_outputRes = 0x10000;
+  CurveDataSample(&c->c, &c->csample);
+  dt_draw_curve_smaple_values(c, min, max, res, x, y);
+}
+
+static inline void dt_draw_curve_calc_values_V2_nonperiodic(dt_draw_curve_t *c, const float min, const float max,
+                                                            const int res, float *x, float *y)
+{
+  c->csample.m_samplingRes = res;
+  c->csample.m_outputRes = 0x10000;
+  CurveDataSampleV2(&c->c, &c->csample);
+  dt_draw_curve_smaple_values(c, min, max, res, x, y);
+}
+
+static inline void dt_draw_curve_calc_values_V2_periodic(dt_draw_curve_t *c, const float min, const float max,
+                                                         const int res, float *x, float *y)
+{
+  c->csample.m_samplingRes = res;
+  c->csample.m_outputRes = 0x10000;
+  CurveDataSampleV2Periodic(&c->c, &c->csample);
+  dt_draw_curve_smaple_values(c, min, max, res, x, y);
+}
+
+static inline void dt_draw_curve_calc_values_V2(dt_draw_curve_t *c, const float min, const float max,
+                                                const int res, float *x, float *y, const _Bool periodic)
```

Comment:
_Bool should be gboolean. This makes clang failing.

Reply:
Why? `_Bool` is a standard data type of the C programming language just like `int`. See https://en.wikipedia.org/wiki/C_data_types It is used in darktable sources at other places without causing any problems, see `blend.c` for example.

Reply:
Maybe you should include `<stdbool.h>` then

Reply:
No, please use gboolean as this is what is used in other routine in draw.h.

Reply:
@rabauke : Why ?

Because it breaks the automatic test on Travis ! And because for consistency we want to use gboolean. _Bool is only used in some part (10 occurrences) of blend.c and some instance on Lua interface, we don't want to spread its use.

Reply:
I investigated this issue in some detail. It is an interoperability problem between C and C++. Compilation error is caused when `src/iop/introspection_lens.cc` includes `src/gui/draw.h`. I will fix this as soon as possible.

Reply:
I replaced `_Bool` by `gboolean` as requested. Code compiles also with Clang now.